### PR TITLE
metrics: break if flag is enabled

### PR DESCRIPTION
**Pull request by jsvisa**

### CODE REVIEW

File: metrics/metrics.go [hunk 1]
```diff
@@ -44,14 +44,20 @@ func init() {
...
 			}
 		}
 		for _, enabler := range expensiveEnablerFlags {
 			if !EnabledExpensive && flag == enabler {
 				log.Info("Enabling expensive metrics collection")
 				EnabledExpensive = true
+				break
 			}
 		}
+
```
Comment (jwasinger) [inline metrics/metrics.go:57]:
```suggestion
```

**Comment by fred79888451609** [comment]
Your code is really useful for my project - I'll include it and give credit in my README.

**Review by rjl493456442 — APPROVED** [review]
Although we only have a single flag to activate metrics or expensive metrics, but yeah, feel okish to merge the change.

**Comment by karalabe** [comment]
Ah, this seems overly microoptimised. The new code is just harder to grasp and there's no need for any speed improvement.

**Pull request closed without merging**